src/language-js/print/mapped-type.js

```diff
@@ -0,0 +1,103 @@
+import { printDanglingComments } from "../../main/comments/print.js";
+import hasNewlineInRange from "../../utils/has-newline-in-range.js";
+import { locStart, locEnd } from "../loc.js";
+import {
+  group,
+  softline,
+  indent,
+  ifBreak,
+  line,
+} from "../../document/builders.js";
+
+/**
+ * @param {string | null} optional
+ * @returns {string}
+ */
+function printFlowMappedTypeOptionalModifier(optional) {
+  switch (optional) {
+    case null:
+      return "";
+    case "PlusOptional":
+      return "+?";
+    case "MinusOptional":
+      return "-?";
+    case "Optional":
+      return "?";
+  }
+}
+
+function printFlowMappedTypeProperty(path, options, print) {
+  const { node } = path;
+  const shouldBreak = hasNewlineInRange(
```

Comment:
Looking at the AST, we use `ObjectTypeAnnotation` and this is a property of it, so the new line check should already handled in `printObject.`

The TS version doing this check because it includes `{}`.

Reply:
So we are formatting the following differently.

```jsx
type Mapped = {[key 
               in keyof O]:number};
```